jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -49,13 +50,27 @@ def parse_data(app, file_obj, data_type=None, data_label=None):
         file_name = os.path.basename(file_obj)
 
         with fits.open(file_obj) as hdulist:
-            _parse_hdu(app, hdulist, file_name=data_label or file_name)
+            prihdr = hdulist[0].header
+            telescop = prihdr.get('TELESCOP', '').lower()
+            filetype = prihdr.get('FILETYPE', '').lower()
+            if telescop == 'jwst' and filetype == '3d ifu cube':
+                from glue.core import Data
+                unit = u.Unit(hdulist[1].header.get('BUNIT', 'count'))
+                flux = hdulist[1].data << unit
+                wcs = WCS(hdulist[1].header, hdulist)
+                data = Data(flux=flux, coords=wcs)  # Spectrum1D too slow
```

Comment:
I decided to bypass `Spectrum1D` parsing because of two problems:

1. astropy/specutils#688
2. Even if I wait patiently for it to load, Glue tries to swap something in the data translation phase (`Spectrum1D` -> `Data`) and fails.

This is what I tried (something like that):

```python
sp = Spectrum1D.read(file_obj, format='JWST s3d') 
app.add_data(sp, data_label)
```

Because I did not pass in `Spectrum1D`, but rather native `Glue` object, line list and unit conversion plugins break because they make certain assumption of the data types.